### PR TITLE
Research now requires Toxins/Robotics Access with Doors

### PR DESCRIPTION
**Pull request by JamieD1**
Research now requires Toxins/Robotics Access and now has Doors
Also updates some windows and fixes access on a couple doors.

HoS is now trespassing and can be arrested for going in R&D without cause.

:cl:  
rscadd: Research now requires Toxins/Robotics Access and now has Doors
/:cl:

### CODE REVIEW

**Comment by N3D6** [comment]
Which option did you go with? Can we get a screenshot?

**Comment by maxion12345** [comment]
XiEA4oq
>not even glass airlocks

File: code/game/objects/items/circuitboards/computer_circuitboards.dm [hunk 1]
```diff
@@ -352,6 +352,8 @@
 
 /obj/item/circuitboard/computer/rdconsole/attackby(obj/item/I, mob/user, params)
 	if(I.tool_behaviour == TOOL_SCREWDRIVER)
+		if(build_path == /obj/machinery/computer/rdconsole/production)
```
Comment (Bibby0110) [inline code/game/objects/items/circuitboards/computer_circuitboards.dm:355]: Undocumented change

File: code/modules/research/rdconsole.dm [hunk 2]
```diff
@@ -699,6 +699,15 @@ Nothing else in the console has ID requirements.
 /obj/machinery/computer/rdconsole/proc/machine_icon(atom/item)
 	return icon2html(initial(item.icon), usr, initial(item.icon_state), SOUTH)
 
+/obj/machinery/computer/rdconsole/proc/can_research(mob/user)
+	if(locked == FALSE)
```
Comment (Bibby0110) [inline code/modules/research/rdconsole.dm]:
```suggestion
	if(!locked)
```
Consoles are by default unlocked. You can’t research if it’s locked IIRC

File: code/modules/research/rdconsole.dm [hunk 3]
```diff
@@ -699,6 +699,15 @@ Nothing else in the console has ID requirements.
...
+		return TRUE
+	var/obj/item/card/id/C = user.get_idcard(TRUE)
```
Comment (Bibby0110) [inline code/modules/research/rdconsole.dm]: Just check allowed(user)

**Comment by JamieD1** [comment]
> XiEA4oq
> 
> > not even glass airlocks

Fixed

**Review by Bibby0110 — CHANGES_REQUESTED** [review]

File: code/modules/research/rdconsole.dm (hunk 2 above)
Comment (JamieD1) [reply code/modules/research/rdconsole.dm]: Its for an emag check

File: code/game/objects/items/circuitboards/computer_circuitboards.dm (hunk 1 above)
Comment (JamieD1) [reply code/game/objects/items/circuitboards/computer_circuitboards.dm:355]: Bugfix

**Comment by ToasterBiome** [comment]
Why though?

**Comment by Cartlord** [comment]
Why not use the large airlock? It's not a Clerk-only thing, ruins use it too.

**Comment by Runian** [comment]
What's this suppose to stop? You can walk into the experiment room for the research console and you can go to the testing lab to access the secondary protolathe unless the airlock access has been changed.

**Review by ToasterBiome — CHANGES_REQUESTED** [review]
honestly jst do the machinery one the map change sucks ass

File: code/game/objects/items/circuitboards/computer_circuitboards.dm (hunk 1 above)
Comment (Bibby0110) [reply code/game/objects/items/circuitboards/computer_circuitboards.dm:355]: What bug?

**Comment by N3D6** [comment]
> Why not use the large airlock? It's not a Clerk-only thing, ruins use it too.

It’s ugly and buggy

**Comment by SomeguyManperson** [comment]
super ultra hugbox extreme 2

File: code/game/objects/items/circuitboards/computer_circuitboards.dm (hunk 1 above)
Comment (SomeguyManperson) [reply code/game/objects/items/circuitboards/computer_circuitboards.dm:355]: if you're going to be making a bugfix atomize it don't shove it into another pr

**Comment by maxion12345** [comment]
> > Why not use the large airlock? It's not a Clerk-only thing, ruins use it too.
> 
> It’s ugly and buggy

everyone out here says it buggy yet I can't find a single mf to tell me how

**Comment by maxion12345** [comment]
> super ultra hugbox extreme 2

TFW xantam has you fix the """"bug""""" with the engineering console

**Pull request merged**